torch/csrc/jit/tensorexpr/schedule.cpp

```diff
@@ -392,6 +392,237 @@ ScheduleObject* ScheduleNode::CloneScheduleObject(ScheduleObject* object) {
   return new_object;
 }
 
+
+class Vectorizer : public IRMutator {
+  public:
+  Stmt* vectorize(const For* v) {
+    Stmt* body = v->body();
+    const Var* var = v->var();
+    const Expr* start = v->start();
+    const Expr* stop = v->stop();
+
+    const IntImm* start_imm = dynamic_cast<const IntImm*>(start);
+    const IntImm* stop_imm = dynamic_cast<const IntImm*>(stop);
+    if (!start_imm) {
+      throw "Can't vectorize due to non-constant loop start!";
```

Comment:
Throwing strings seems kind of sketch.  What's pytorch convention for exceptions?  Should we just use `std::runtime_error`?